var/spack/repos/builtin/packages/raja/package.py

```diff
@@ -56,10 +57,13 @@ class Raja(CachedCMakePackage, CudaPackage, ROCmPackage):
     depends_on('blt@0.4.0:', type='build', when='@0.13.0')
     depends_on('blt@0.3.6:', type='build', when='@:0.12.0')
 
-    depends_on('camp@0.2.2', when='@0.14.0:')
-    depends_on('camp@0.1.0', when='@0.12.0:0.13.0')
+    depends_on('camp@0.2.2', when='@0.14.0')
+    depends_on('camp@0.1.0', when='@0.10.0:0.13.0')
+    depends_on('camp@2022.03.0:', when='@2022.03.0:')
+    conflicts('camp', when='@:0.9.0')
```

Comment:
This should be 

```
conflicts('^camp', when='@:0.9.0')
```

since `conflicts('camp')` would prevent `camp` from appearing anywhere in the concretization (even as a dependency of some other package).

Overall, I think for the desired output, it would be more-straightforward to say `depends_on("raja@0.12:")` in dray - could you do that?